hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/model/ArtifactUpload.java

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2015 Bosch Software Innovations GmbH and others.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.eclipse.hawkbit.repository.model;
+
+import java.io.InputStream;
+
+import javax.validation.constraints.NotEmpty;
+import javax.validation.constraints.NotNull;
+
+import org.eclipse.hawkbit.repository.RegexCharacterCollection;
+import org.eclipse.hawkbit.repository.RegexCharacterCollection.RegexChar;
+import org.hibernate.validator.constraints.SafeHtml;
+
+/**
+ * Use to create a new artifact.
+ *
+ */
+public class ArtifactUpload {
+
+    public static final RegexCharacterCollection ILLEGAL_FILENAME_CHARACTERS = new RegexCharacterCollection(
```

Comment:
Move to `AbstractFileTransferHandler` since it is only used within the two subclasses

Reply:
ArtifactUpload is also used in MgmtSoftwareModuleResource when calling ArtifactManagement.create during an upload via mgmt-api